### ISSUE TITLE
Is the "test_accuracy" in test_metrics.csv in output files already the mean accuracy of Leave-One-Subject-Out CV?

### ISSUE THREAD

**Issue by mhzhang96** [wenhui0206/NeuroGPT#7]
Hi,

As in subject, I'm curious about is the "test_accuracy" in test_metrics.csv in output files already the mean accuracy out of 9 subjects using Leave-One-Subject-Out CV?

Thank you.

Also I think in train_gpt.py, line 155:

> test_dataset = train_dataset

seems confusing, is it a mistake?

**Comment by henrisuurorg** [comment]
> test_dataset = train_dataset

@chocooni Did you ever find out if this was intentional? It's confusing for me as well. 

Any comment @wenhui0206 ?

**Comment by wenhui0206** [comment]
Hi, sorry for the confusion. You can ignore the test_dataset here. The real testing phase (fine-tuning) is a separate run which is defined in  scripts/finetune.sh.

**Comment by gjoyet** [comment]
Hello

I have a question about this, too. It seems like in the downstream task (i.e. `config["training_style"] == 'decoding'`), we load a training and a test set which contain data from disjoint sets of subjects. However, with the line `test_dataset = train_dataset`, the variable `test_dataset` now also contains the training data. 

However, in train_gpt.py, on line 225-226, we then have:

```
    if test_dataset is not None:
        test_prediction = trainer.predict(test_dataset)
```

so it seems like we evaluate the model on the set which actually contains the training data. Is this voluntary? If so, can you explain the idea behind it?

Many thanks in advance
